### PR TITLE
Feature/search component

### PR DESCRIPTION
**Pull request by mattdean-digicatapult**
Search component. Renders a mini form containing a search input and a button for submit. The `onSubmit` handler will be called either after a debounce (default 250ms) automatically or on form submit. Configurable colors (not yet tied to a theme). Story allows configurable container and props and display of the `onSubmit` action.

Screen-Recording-2022-12-14-at-0

### CODE REVIEW

File: src/Drawer/__snapshots__/Drawer.spec.js.snap [hunk 1]
```diff
@@ -405,6 +413,10 @@ exports[`Drawer with overrides 1`] = `
   border: 0;
 }
 
+.c1:hover {
```
Comment (n3op2) [inline src/Drawer/__snapshots__/Drawer.spec.js.snap:416]: nice, like that. I mean 'improvement'

File: src/Search/Search.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      <Search
+        placeholder={args.placeholder}
+        color={args.color}
+        background={args.background}
+        onSubmit={action('submit')}
+      />
+    </SearchHost>
+  )
+}
+export const Default = DefaultStoryTemplate.bind({})
```
Comment (n3op2) [inline src/Search/Search.stories.tsx:28]: i'm not using template so not sure maybe i should create a story to update the one that don\t and from now use it? 

```js
export const Default: Story<typeof AppBar> = (props) => (
  <AppBar {...props}>
    <AppBar.Item active={true}>item one</AppBar.Item>
    <AppBar.Item>item two</AppBar.Item>
  </AppBar>
)
```

File: src/Search/__snapshots__/Search.spec.js.snap [hunk 3]
```diff
@@ -0,0 +1,254 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Test default 1`] = `
+.c0 {
+  display: -webkit-box;
```
Comment (n3op2) [inline src/Search/__snapshots__/Search.spec.js.snap:5]: like styled components for that, generating for each browser

File: src/Search/index.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  background = 'transparent',
+  debounce = 250,
+  onSubmit,
+}) => {
+  const [search, setSearch] = useState<string | null>(null)
+  const [hasSubmitted, setHasSubmitted] = useState(true)
+  const buttonRef = useRef<HTMLButtonElement | null>(null)
+
+  const handleSearchChange: React.ChangeEventHandler<HTMLInputElement> =
+    useCallback(
```
Comment (n3op2) [inline src/Search/index.tsx:36]: never used this method. interesting

File: src/Search/index.tsx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    transform: rotateZ(40deg);
+    transform-origin: bottom right
+  }
+
+  :hover {
+    cursor: pointer;
+  }
+`
+
+export default Search
```
Comment (n3op2) [inline src/Search/index.tsx:161]: like the idea that we are no longer creating a new file just for two lines of code

**Review by n3op2 — APPROVED** [review]

File: src/Search/Search.stories.tsx (hunk 2 above)
Comment (mattdean-digicatapult) [reply src/Search/Search.stories.tsx:28]: I just took that from the stroybook docs 🤷‍♂️

File: src/Search/index.tsx (hunk 4 above)
Comment (mattdean-digicatapult) [reply src/Search/index.tsx:36]: if you define a function in a component you should probably be using `useCallback`. It prevents garbage collection spam by reducing the need to constantly recreate a function on new renders

**Pull request merged**